DigitalLearningSolutions.Data/Services/EmailService.cs

```diff
@@ -29,6 +31,55 @@ public class EmailService : IEmailService
         }
 
         public void SendEmail(Email email)
+        {
+            var mailConfig = GetMailConfig();
+
+            MimeMessage message = CreateMessage(email, mailConfig.MailSenderAddress);
+
+            try
+            {
+                using var client = smtpClientFactory.GetSmtpClient();
+                client.Timeout = 10000;
+                client.Connect(mailConfig.MailServerAddress, mailConfig.MailServerPort);
+
+                client.Authenticate(mailConfig.MailServerUsername, mailConfig.MailServerPassword);
+
+                client.Send(message);
+                client.Disconnect(true);
+            }
+            catch (Exception error)
+            {
+                logger.LogError(error, "Sending an email has failed");
+            }
+        }
+
+        public void SendEmails(IEnumerable<Email> emails)
+        {
+            var mailConfig = GetMailConfig();
+
+            using var client = smtpClientFactory.GetSmtpClient();
+            client.Timeout = 10000;
+            client.Connect(mailConfig.MailServerAddress, mailConfig.MailServerPort);
+            client.Authenticate(mailConfig.MailServerUsername, mailConfig.MailServerPassword);
```

Comment:
What happens if connection to email client fails (is removing the try/catch intentional)?

Reply:
I moved the try-catch inside so it could deal with emails individually, but there also needs to be one outside for this case, whoops.